src/main/java/bowling/domain/state/Bonus.java

```diff
@@ -0,0 +1,65 @@
+package bowling.domain.state;
+
+import bowling.annotations.ForUI;
+import bowling.domain.KnockedPinCount;
+import bowling.domain.PinCounts;
+import bowling.domain.Score;
+
+public class Bonus extends AbstractFinished {
+    private static final int SPARE_BONUS_COUNT = 1;
+    private static final int STRIKE_BONUS_COUNT = 2;
+
+    private final int bonusCount;
+    private final State previous;
+
+    public Bonus(PinCounts pinCounts, int bonusCount, State state) {
+        super(pinCounts);
+        this.bonusCount = bonusCount;
+        this.previous = state;
+    }
+
+    public static Bonus ofSpare(PinCounts pinCounts, Spare spare) {
+        return new Bonus(pinCounts, SPARE_BONUS_COUNT - ONE, spare);
+    }
+
+    public static Bonus ofStrike(PinCounts pinCounts, Strike strike) {
+        return new Bonus(pinCounts, STRIKE_BONUS_COUNT - ONE, strike);
+    }
+
+    @Override
+    public State bowl(int pinCount) {
+        return new Bonus(pinCounts.knockOut(pinCount), bonusCount - ONE, previous);
+    }
+
+    @Override
+    public boolean isFinished() {
+        return bonusCount == ZERO;
+    }
+
+    @Override
+    public boolean hasBonus() {
+        return false;
+    }
+
+    @Override
+    public Score makeScore() {
+        int sum = getValues().stream().map(KnockedPinCount::value)
+                .reduce(ZERO, Integer::sum)
+                .intValue();
```

Comment:
```suggestion
        int sum = getValues().stream().map(KnockedPinCount::value)
                .reduce(ZERO, Integer::sum);
```
언박싱을 하지 않아도 될 것 같아요 ~ :)